packages/test-project/src/components/DynamicValidations.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+  <div class="SimpleForm">
+    <div class="">
+      <label>x</label>
+      <input
+        v-model="x"
+        type="number"
+      >
+    </div>
+    <div class="">
+      <label>y</label>
+      <input
+        v-model="v$.y.$model"
+        type="text"
+      >
+    </div>
+    <div class="">
+      <label>a</label>
+      <input
+        v-model="a"
+        type="number"
+      >
+    </div>
+    <div class="">
+      <label>y</label>
+      <input
+        v-model="$v.b.$model"
+        type="text"
+      >
+    </div>
+    <button @click="$v.$touch">
+      touch
+    </button>
+    <div style="background: rgba(53, 219, 53, 0.62); color: #ff9090; padding: 10px 15px">
+      <p
+        v-for="(error, index) of $v.$errors"
+        :key="index"
+        style="padding: 0; margin: 5px 0"
+      >
+        {{ error.$message }}
+      </p>
+    </div>
+    <div style="background: rgba(219, 53, 53, 0.62); color: #ff9090; padding: 10px 15px">
+      <p
+        v-for="(error, index) of v$.$errors"
+        :key="index"
+        style="padding: 0; margin: 5px 0"
+      >
+        {{ error.$message }}
+      </p>
+    </div>
+    <pre>{{ v$ }}</pre>
+    <pre>{{ $v }}</pre>
+  </div>
+</template>
+
+<script>
+import { ref, computed } from 'vue'
+import { VuelidateMixin, useVuelidate } from '@vuelidate/core'
+import { minLength, required } from '@vuelidate/validators'
+
+export default {
+  name: 'SimpleForm',
+  mixins: [VuelidateMixin],
+  setup () {
+    const x = ref(5)
+    const y = ref('a')
+
+    const rules = computed(() => {
```

Comment:
@shentao do you think it would work if a specific property has a dynamic rule set?

```js
{
  y: computed(() => { x.value >= 6 ? { required } : { somethingElse } })
  x: { required }
}
```

Reply:
Interesting! The number of possibilities here is getting quite overwhelming. I think it might break the current implementation. We either try to accommodate for that or we just recommend that people make the whole rules object a computed if they want to have dynamic rules? WDYT?

Reply:
I think we should try to see if we can support this with minimal changes. If not, we recommend that they make the entire validation object dynamic. 

Depends on the case really. Composition API gives too much freedom 😆

Reply:
Thats exactly how I feel about it. We need to review the whole flow again and see where we can normalize early so that we remove some conditionals later on.